### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by PakhiAgarwal**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -11,7 +18,7 @@ endif ()
 # for Windows only
 if (WIN32)
   set(CMAKE_CXX_FLAGS_RELEASE "-DNOMINMAX ${CMAKE_CXX_FLAGS_RELEASE} /MT")
-  set(CMAKE_CXX_FLAGS_DEBUG "-DNOMINMAX ${CMAKE_CXX_FLAGS_DEBUG} /MTd")
+  set(CMAKE_CXX_FLAGS_DEBUG "-DNOMINMAX ${CMAKE_CXX_FLAGS_DEBUG} /MDd")
```
Comment (rjzupkoii) [inline CMakeLists.txt:21]: Add some documentation as to why the switch needed to be changed.

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,6 +1,13 @@
-cmake_minimum_required(VERSION 3.2)
+cmake_minimum_required(VERSION 3.10)
```
Comment (rjzupkoii) [inline CMakeLists.txt:1]: Not sure we want to go down a version like this, @merlinvn any thoughts?

File: main.cpp [hunk 3]
```diff
@@ -0,0 +1,11 @@
+//
+// Created by pua66 on 2/12/2021.
```
Comment (rjzupkoii) [inline main.cpp:2]: I think this can safely be removed from the main code base.

**Review by rjzupkoii — CHANGES_REQUESTED** [review]
@merlinvn Anything to add to the review?

**Comment by PakhiAgarwal** [comment]
Is there any way to simplify the `CmakeLists.txt`? Do I need to add all the dependent header files and source file everytime?

**Comment by rjzupkoii** [comment]
@PakhiAgarwal It's possible to simplify if you have source directories, recall some of the documentation you've been reading, and take some time to read through the `CMakeLists.txt` files are setup for the main simulation.

File: CMakeLists.txt (hunk 2 above)
Comment (merlinvn) [reply CMakeLists.txt:1]: Basicly, no need to go up to Version 3.10

File: CMakeLists.txt [hunk 4]
```diff
@@ -1,6 +1,13 @@
...
+set(CMAKE_TOOLCHAIN_FILE
+        ${CMAKE_CURRENT_SOURCE_DIR}/ext/vcpkg/scripts/buildsystems/vcpkg.cmake  CACHE
+        STRING "Vcpkg toolchain file")
+
 project(POMS)
 #set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/cmake/Modules")
+set(CMAKE_CXX_STANDARD 20)
```
Comment (merlinvn) [inline CMakeLists.txt:8]: I don't think that we need C++20, this line is from my template project

File: CMakeLists.txt [hunk 5]
```diff
@@ -1,6 +1,13 @@
...
+set(CMAKE_CXX_STANDARD_REQUIRED True)
```
Comment (merlinvn) [inline CMakeLists.txt:9]: This is not needed as well

File: CMakeLists.txt [hunk 6]
```diff
@@ -68,3 +75,6 @@ endif ()
 #    DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/bin/hello_2
 #    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
 #)
+
+
+add_subdirectory(test)
```
Comment (merlinvn) [inline CMakeLists.txt:80]: we already add_subdirectory(test), all you just need is to turn on the flag BUILD_TESTS when using cmake to generate the project build

File: test/CMakeLists.txt [hunk 7]
```diff
@@ -1,36 +1,30 @@
...
-#
-#install(TARGETS ${PROJECT_TEST_NAME} DESTINATION ${PROJECT_SOURCE_DIR}/bin)
-#install(FILES ${PROJECT_SOURCE_DIR}/misc/input.yml DESTINATION ${PROJECT_SOURCE_DIR}/bin)
+# 'Google_test' is the subproject name
+find_package(fmt CONFIG REQUIRED)
+find_package(GTest CONFIG REQUIRED)
+find_package(GSL REQUIRED)
+#project(Google_tests)
+
+set(SRC MDC/ModelDataCollector.cpp MDC/ModelDataCollector.h Spatial/Location.h
```
Comment (merlinvn) [inline test/CMakeLists.txt:7]: This set is redundent

File: src/Reporters/TACTReporter.cpp [hunk 8]
```diff
@@ -3,7 +3,7 @@
 //
 
 #include <Model.h>
-#include <MDC/ModelDataCollector.h>
+#include "MDC/ModelDataCollector.h"
```
Comment (merlinvn) [inline src/Reporters/TACTReporter.cpp:6]: You don't need to commit this file

File: CMakeLists.txt (hunk 1 above)
Comment (merlinvn) [reply CMakeLists.txt:21]: Yeah, this was added long time ago for Gtest experiments works, and they might changed / fixed the machanism. So you can try to remove those 2 above lines to see if it work or not, if it work then we can simply just comment and document it.

**Pull request closed without merging**